pos_tare/static/src/js/Screens/ProductScreen/ProductScreen.js

```diff
@@ -0,0 +1,167 @@
+odoo.define("pos_tare.screens", function (require) {
+    "use strict";
+    const ProductScreen = require("point_of_sale.ProductScreen");
+    const Registries = require("point_of_sale.Registries");
+    const {useBarcodeReader} = require("point_of_sale.custom_hooks");
+
+    const TareProductScreen = (ProductScreen) =>
+        class extends ProductScreen {
+            constructor() {
+                super(...arguments);
+                useBarcodeReader({
+                    // We add the tare action
+                    tare: this._barcodeTareAction,
+                });
+            }
+
+            async _barcodeTareAction(code) {
+                var last_orderline = this.currentOrder.get_last_orderline();
+                if (last_orderline) {
+                    last_orderline.set_tare(code.value, true);
+                }
+            }
+
+            async _getAddProductOptions(product) {
```

Comment:
rewriting all this function is not a good option.  
I think you can return weight + tare in the payload of the "ScaleScreen", and then overload _getAddProductOptions just to split the quantity field into the two weight and tare. What do you thing ?

Reply:
Hello @legalsylvain and thanks for your comment.
As I remember I really tried not to do that but I didn't have the choice. I cannot tell you why exactly as I currently work on other projects and haven't touch this for months. If you are sure of you, you can maybe propose some changes and I'll gladly take a look (and test).